### PR TITLE
Outline for language core and extensions

### PR DESCRIPTION
**Pull request by Zimmi48**
This PR adds comments to the Core language and Language extensions chapters to give the planned outline. To be discussed.

cc @mattam82 @jfehrle and anyone interested

cf. coq/ceps#43

### CODE REVIEW

**Comment by SkySkimmer** [comment]
What's the difference between "core" and "extension"?

**Comment by Zimmi48** [comment]
The core language is meant to reflect what the kernel understands, whereas the language extensions are entirely removed at elaboration time. Cf. the explanations in coq/ceps#43.

File: doc/sphinx/language/core/index.rst [hunk 1]
```diff
@@ -30,10 +30,67 @@ will have to check their output.
 .. toctree::
    :maxdepth: 1
 
+   .. The two chapters Gallina specification language and CIC are
+      planned for removal.  Eventually, the present "Core language"
```
Comment (jfehrle) [inline doc/sphinx/language/core/index.rst:34]: The Gallina spec chapter file goes away, but the material will be moved.
Will any of CIC be dropped entirely?  Also, expect there will be more rewriting here to make all the formulas more accessible.

File: doc/sphinx/language/core/index.rst [hunk 2]
```diff
@@ -30,10 +30,67 @@ will have to check their output.
...
+      Gallina extensions chapter.
+
    ../gallina-specification-language
    ../cic
+
+   .. Comments are used to show upcoming sections (not yet split out).
+
+   .. Lexical conventions, how to read the grammars in the manual,
+      presentations of the basic notions: terms, commands, tactics,
+      attributes and options.
```
Comment (jfehrle) [inline doc/sphinx/language/core/index.rst:46]: Indentation is your friend and can show the planned hierarchy.  Would be more readable as a phrase outline, e.g.

```
Lexical conventions
  how to read the grammars in the manual
  basic notions
    terms
    commands
    tactics
    attributes
    options
```

File: doc/sphinx/language/core/index.rst [hunk 3]
```diff
@@ -30,10 +30,67 @@ will have to check their output.
...
+
+   .. Terms, typing and conversion (only basic constructs:
+      abstraction, product, application, type cast, brief mention of
+      sorts).
+
+   .. Assumptions
```
Comment (jfehrle) [inline doc/sphinx/language/core/index.rst:52]: What goes under "Assumptions"?

File: doc/sphinx/language/core/index.rst [hunk 4]
```diff
@@ -30,10 +30,67 @@ will have to check their output.
...
+
+   .. Definitions (vernac) and local definitions (let in).  Includes
+      opaque definitions and a brief introduction to the interactive
+      proof mode, in the spirit of CEP #42.
```
Comment (jfehrle) [inline doc/sphinx/language/core/index.rst:56]: Wouldn't interactive proof mode naturally go in the Writing Proofs chapter?  IIUC it's not a core language feature.

File: doc/sphinx/language/core/index.rst [hunk 5]
```diff
@@ -30,10 +30,67 @@ will have to check their output.
...
+      or in the Libraries chapter or merged with the section on
+      conversion above?
+
+   .. The next section would be about sorts, would include the current
+      addendum chapter about SProp, but also explain clearly
+      Set and Prop, imprecativity, the -imprecative-set option.
+
    ../../addendum/sprop
+
+   .. The next section would be about universes, including how to
```
Comment (jfehrle) [inline doc/sphinx/language/core/index.rst:80]: "would be" makes it sound like the section doesn't exist.  The comment describes planned changes/additions, right?  Maybe "will add"?

Same comment on 2 other "would be"s, one above and below

File: doc/sphinx/language/extensions/index.rst [hunk 6]
```diff
@@ -16,12 +16,43 @@ language presented in the :ref:`previous chapter <core-language>`.
 .. toctree::
    :maxdepth: 1
 
+   .. The two chapters at the top and the bottom (Gallina extensions
+      and Vernacular commands) are planned for removal.  Eventually,
```
Comment (jfehrle) [inline doc/sphinx/language/extensions/index.rst:20]: removal -> split up and integrate the material elsewhere

**Review by jfehrle — APPROVED** [review]
Thanks for creating this.  Not sure if it will be super helpful for grasping the overall structure.  I suspect that as we proceed and I read updated chapters closely I'll have a lot more thoughts about the ordering.  A lot of the existing material I don't understand well enough and/or I gave up trying to read.

Ordering is more important for things we expect people to read through in a focused way from beginning to end.  For reference material, I expect the most common use will be people jumping in from the index/TOC and following hyperlinks.  Sometime I like to skim an entire manual or chapter to get a sense of what's in it, but I don't need perfect ordering for that.

File: doc/sphinx/language/core/index.rst (hunk 1 above)
Comment (Zimmi48) [reply doc/sphinx/language/core/index.rst:34]: I don't think any of the current material should be removed. Making it more accessible shouldn't mean making it less complete. Besides, with the structure for each section that we proposed in the CEP, it should be OK that in each section, there are some explanations that are less accessible than others:

>Topics are gathered in a single place where they are explained in depth, starting with a high level view, with lots of examples, then a more exhaustive presentation aimed at the experts, to finish with theoretical and implementation considerations that hint at how to improve the actual code, what are the limitations, and how to contribute.

File: doc/sphinx/language/core/index.rst (hunk 3 above)
Comment (Zimmi48) [reply doc/sphinx/language/core/index.rst:52]: The current "Assumptions" section of the Gallina chapter.

File: doc/sphinx/language/core/index.rst (hunk 4 above)
Comment (Zimmi48) [reply doc/sphinx/language/core/index.rst:56]: Yes, I just meant briefly write that it exists and that link to the chapter where it is docummented.

**Comment by Zimmi48** [comment]
Superseded by #12239

**Pull request closed without merging**